src/models/cart.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable max-len */
+const mongoose = require('mongoose');
+
+const cartSchema = mongoose.Schema({
+  username: { type: String, required: true },
+  // object of dynamic keys. As shown in the sample below, 1 is the productId.
+  // Cannot define dynamic keys in mongo schema
+  products: { type: Object, default: {} },
+}, { minimize: false }); // {1: {sizeSelected: 's', amount: 3 }, {sizeSelected: 'm', amount: 6 },..},
+
+cartSchema.statics.getCart = async (username) => {
+  const cart = await cartModel.findOne({ username }).exec();
+
+  return cart;
+};
+
+cartSchema.statics.updateCart = async (username, updatedProducts) => {
```

Comment:
Await is really not needed in all these statements. 

The places where I use this function has the await. I believe its fine that its returning a promise... the await is handled in the actual controller. For example, in controllers/update-cart.js. We have the await for this function `    return await Cart.updateCart(userId, updatedProducts);` which will wait for this promise to resolve.  

Okay this is just a question... shouldn't the `try/catch` happen on the block of code that is using this function? I feel like it might be too repetitive adding try/catch everything. Like lets look at controllers/update-cart.js again. There is a `try/catch` there and adding another `try/catch` in this function `cartSchema.statics.updateCart = async (username, updatedProducts) => {` seem kinda redundant IMO. But I would like to hear your opinion

Reply:
You're absolutely correct. `async` functions are themselves promises so it is redundant to use `return await <some-method>` within an function. You can update the code to simple return the promise `return Cart.updateCart(userId, updatedProducts)`

Ideally, every method should be wrapped in a `try/catch` block _unless_ you want to defer the error handling up to a method that is earlier in the call chain. The benefit of wrapping all methods in a `try/catch` is that when a method throws an error you stack trace will indicate exactly which function and line threw the error. If it is deferred to the calling method, the stack trace will show that the calling method threw the error, which is not always the case.

All that being said, when an error is thrown, JavaScript stores all calling methods in the stack trace so you could technically read/parse that to find where your error is sourced. However, this is tedious and requires you printing and reading the entire stack trace instead of a one-line error message which includes the filepath, function and line number where the error originated.